jsx/Breadcrumbs.js

```diff
@@ -14,25 +16,106 @@
  */
 class Breadcrumbs extends React.Component {
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      displayCount: 0
+    };
+
+    this.checkScreenSize = this.checkScreenSize.bind(this);
+  }
+
+  componentWillMount() {
+    this.checkScreenSize();
+    if (typeof window !== 'undefined')
+      window.addEventListener('resize', this.checkScreenSize);
+  }
+
+  componentWillUnmount() {
+    if (typeof window !== 'undefined')
+      window.removeEventListener('resize', this.checkScreenSize);
+  }
+
+  checkScreenSize() {
+    // Used to check to current window size and
+    // sets the number of breadcrumbs to show
+    var windowWidth = window.innerWidth;
+    var displayCount = 4;
+
+    if (windowWidth <= 425) {
+      displayCount = 1;
+    } else if (windowWidth <= 500) {
+      displayCount = 2;
+    } else if (windowWidth <= 767) {
+      displayCount = 3;
+    } else if (windowWidth > 885 && windowWidth <= 1350) {
+      displayCount = 3;
+    } else if (windowWidth > 1500) {
+      displayCount = 5;
+    }
+
+    this.setState({
+      displayCount: displayCount
+    });
+  }
+
   render() {
     const baseURL = this.props.baseURL;
-    const breadcrumbs = this.props.breadcrumbs.map(function(element, i) {
-      const url = baseURL + element.query;
-      const crumb = (
-        <a key={'crumb_' + i} href={url} className="btn btn-primary">
-          <div>
-            {element.text}
-          </div>
-        </a>
+    const breadcrumbs = [];
+    const dropdown = [];
+    let breadcrumbDropdown;
+    let url = "";
+    let element = {};
+
+    // Loop through breadcrumbs either adding them to drop down
+    // or displaying them
+    for (var i = 0; i < this.props.breadcrumbs.length; i++) {
```

Comment:
Since this is an ES6 class, it should not contain any `var`s. So far the consensus is to use `const` wherever possible (including arrays and objects) and `let` if the variable is explicitly reassigned later in the code.